src/services/web3/approval/index.ts

```diff
@@ -1,16 +1,22 @@
 import { Token } from 'services/observables/tokens';
 import { web3, writeWeb3 } from 'services/web3';
 import BigNumber from 'bignumber.js';
-import { bancorNetwork$ } from 'services/observables/contracts';
+import {
+  bancorNetwork$,
+  exchangeProxy$,
+  liquidityProtection$,
+} from 'services/observables/contracts';
 import { take } from 'rxjs/operators';
 import { user$ } from 'services/observables/user';
 import {
   NULL_APPROVAL_CONTRACTS,
   UNLIMITED_WEI,
 } from 'services/web3/approval/constants';
-import { ethToken } from 'services/web3/config';
+import { ethToken, getNetworkVariables } from 'services/web3/config';
 import { expandToken } from 'utils/formulas';
 import { Token__factory } from '../abis/types';
+import { ApprovalContract } from 'hooks/useApproveModal';
```

Comment:
this is a dependency between service and UI - where it should be the other way around. If you want to address common enums move them to a better place